crates/red_knot_python_semantic/src/semantic_index/expression.rs

```diff
@@ -8,6 +8,18 @@ use salsa;
 /// An independently type-inferable expression.
 ///
 /// Includes constraint expressions (e.g. if tests) and the RHS of an unpacking assignment.
+///
+/// ## Module local type
+/// This type should not be used as part of any cross module API because
+/// it holds a reference to the AST node and range offset changes
+/// then propagate through all usages and deserialization requires
+/// reparsing the entire module.
```

Comment:
```suggestion
/// ## Module-local type
/// This type should not be used as part of any cross-module API because
/// it holds a reference to the AST node. Range-offset changes
/// then propagate through all usages, and deserialization requires
/// reparsing the entire module.
```